### PR TITLE
adding FlowRunHistory

### PR DESCRIPTION
**Pull request by igalmk**
main:
-added flow history, which is created after every flow run.
the last flow history is also deleted after each new run, so it's needs to be stored in Stepper.
the method: showMinimalFlowHistory is used to show the first info from a list of other histories
the method: showExtensiveFlowHistory is used to show all flow runs history when chosen
note:flow histories need to be stored/presented to user in order where the most recent are first

minor:
-updated time stamps format for logs
-fixed some of step children's summary line update in case of succes/failure/warning
-added method for clearing data members of all steps in case of new flow run
note:i don't really where to place this mehod ,at least it needs to be called outside of Flow
-fixed getPresentableString for mapping DataType

### CODE REVIEW

File: StepperEngine/src/Flow/FlowRunHistory.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+public class FlowRunHistory {
+    private String flowId;
+    private String flowName;
+    private Flow.Status status;
+    private double runTime;
+    private String timeStamp;
+
+    private String freeInputsHistory;
+    private String outputsHistory;
+    private String stepsHistory;
```
Comment (ofirlevy1) [inline StepperEngine/src/Flow/FlowRunHistory.java:15]: I think these are fine for now, but maybe not for the next assignment with the GUI - we might need to change it then.

for example - what if the GUI needs to present "freeInputHistory" not as a whole string, but divided into different inputs (and maybe different fields of each input) in different GUI components (i.e text boxes)

Comment (ofirlevy1) [inline StepperEngine/src/Flow/FlowRunHistory.java:15]: also, what about get() methods for those Strings?

File: StepperEngine/src/Steps/CollectFilesInFolderStep.java [hunk 2]
```diff
@@ -144,4 +144,11 @@ public static int getStepStartUpCount() {
     public static double getStepAvgDuration() {
         return stepAvgDuration;
     }
+
+    public void clearDataMembers(){
+        this.filesList.setData(null);
```
[the remark this answers is not in the document]
Comment (igalmk) [reply StepperEngine/src/Steps/CollectFilesInFolderStep.java:150]: there might be case where a data member in a certein step, will not be updated, thus his old data from previous run will be deliverd to the next step/formal output

File: StepperEngine/src/Flow/Flow.java [hunk 3]
```diff
@@ -311,6 +312,10 @@ public void execute(){
         runTime= Duration.between(start,finish).toMillis();
         calculateAvgRunTime();
         createFlowLog();
+        createFlowHistory();
+        System.out.println(flowRunHistory.showMinimalFlowHistory());
+        System.out.println("\n \n");
+        System.out.println(flowRunHistory.showExtensiveFlowHistory());
```
[the remark this answers is not in the document]
Comment (igalmk) [reply StepperEngine/src/Flow/Flow.java]: yea, they should have been erased in the latest commit

File: StepperEngine/src/Flow/FlowRunHistory.java (hunk 1 above)
Comment (igalmk) [reply StepperEngine/src/Flow/FlowRunHistory.java:15]: you are right, i created them that way so we could progress faster in the current task.
in the following day I will add fields and methods, so FlowRunHistory will be more suited to further tasks

Comment (igalmk) [reply StepperEngine/src/Flow/FlowRunHistory.java:15]: they will be added on next pr

File: StepperEngine/src/Steps/CollectFilesInFolderStep.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ofirlevy1) [reply StepperEngine/src/Steps/CollectFilesInFolderStep.java:150]: are you talking about the "continuation" feature? 

Here you already erase the data completely (setData(null)..) so how can it be used by other steps?

**Pull request merged**